prow/tide/tide.go

```diff
@@ -1124,18 +1125,14 @@ func (c *changedFilesAgent) prChanges(pr *PullRequest) config.ChangedFilesProvid
 		}
 		c.RUnlock()
 
-		// We need to query the changes from GitHub.
-		changes, err := c.ghc.GetPullRequestChanges(
-			string(pr.Repository.Owner.Login),
-			string(pr.Repository.Name),
-			int(pr.Number),
-		)
+		cloneRef := fmt.Sprintf("%s/%s", org, repo)
+		r, err := c.gc.Clone(cloneRef)
 		if err != nil {
-			return nil, fmt.Errorf("error getting PR changes for #%d: %v", int(pr.Number), err)
+			return nil, fmt.Errorf("error cloning repo %s: %v", cloneRef, err)
 		}
-		changedFiles = make([]string, 0, len(changes))
-		for _, change := range changes {
-			changedFiles = append(changedFiles, change.Filename)
+		changedFiles, err = r.DiffThreeDot(string(pr.BaseRef.Name), string(pr.HeadRefOID))
```

Comment:
I don't think `pr.HeadRefOID` will be available locally without fetching the PR branch first? Example: https://github.com/kubernetes/test-infra/blob/dfc1766ef0704616730fdbf591295b957d11dca4/prow/git/git.go#L353-L364

Reply:
Thank you for pointing this out. I did some tests with cloning a github repo and using three dot diff  between master and a PR's headRefOID, and it seems that changes can be obtained this way. Could I be missing some cases? (alternatively, if that seems a better approach, I can simply add a call to `CheckoutPullRequest` before calling on the Diff method).